plutus-ledger-api/src/PlutusLedgerApi/V3/Contexts.hs

```diff
@@ -0,0 +1,577 @@
+  | DRepAlwaysNoConfidence
+  deriving stock (Generic, Haskell.Show, Haskell.Eq)
+  deriving (Pretty) via (PrettyShow DRep)
+
+instance PlutusTx.Eq DRep where
+  {-# INLINEABLE (==) #-}
+  DRepCredential a == DRepCredential a'            = a PlutusTx.== a'
+  DRepAlwaysAbstain == DRepAlwaysAbstain           = Haskell.True
+  DRepAlwaysNoConfidence == DRepAlwaysNoConfidence = Haskell.True
+  _ == _                                           = Haskell.False
+
+data Delegatee
+  = DelegStake V2.PubKeyHash
+  | DelegVote DRep
+  | DelegStakeVote V2.PubKeyHash DRep
+  deriving stock (Generic, Haskell.Show, Haskell.Eq)
+  deriving (Pretty) via (PrettyShow Delegatee)
+
+instance PlutusTx.Eq Delegatee where
+  {-# INLINEABLE (==) #-}
+  DelegStake a == DelegStake a' = a PlutusTx.== a'
+  DelegVote a == DelegVote a' = a PlutusTx.== a'
+  DelegStakeVote a b == DelegStakeVote a' b' =
+    a PlutusTx.== a' PlutusTx.&& b PlutusTx.== b'
+  _ == _ = Haskell.False
+
+{- | There is probably no need for Plutus Scripts to even know about all this data in Anchors,
+but it is here for completeness. So we might wanna omit them, especially since urls are a
+bit tricky, since we can't guarantee their validity, not even that they are utf8-encoded.
+-}
+data Anchor = Anchor
+  { anchorUrl      :: PlutusTx.BuiltinString -- Arbitrary Url
+  , anchorDataHash :: PlutusTx.BuiltinByteString -- Blake2b_256 of some off-chain data
+  }
+  deriving stock (Generic, Haskell.Show, Haskell.Eq)
+
+instance Pretty Anchor where
+  pretty Anchor{..} =
+    vsep
+      [ "anchorUrl:" <+> pretty anchorUrl
+      , "anchorDataHash:" <+> pretty anchorDataHash
+      ]
+
+instance PlutusTx.Eq Anchor where
+  {-# INLINEABLE (==) #-}
+  Anchor a b == Anchor a' b' = a PlutusTx.== a' PlutusTx.&& b PlutusTx.== b'
+
+{- | Note that it no longer is called DCert, since certificates are no longer all about
+delegation, they are also about governance.
+
+Notable changes:
+
+* Pool certificates are unchanged
+* Addition of deposits and refunds. Those values are validated according to respectful
+  PParams. Certificates that where present before get their deposit/refund as
+  optional. This is done for graceful tarnsition, since next era after Conway will require
+  deposit/refund amount
+* New DRep entity
+* New Constitutional Committee entity
+-}
+data TxCert
+  = -- | Register staking credential with an optional deposit amount
+    TxCertRegStaking V2.StakingCredential (Haskell.Maybe V2.Value)
+  | -- | Un-Register staking credential with an optional refund amount
+    TxCertUnRegStaking V2.StakingCredential (Haskell.Maybe V2.Value)
+  | -- | Delegate staking credential to a Delegatee
+    TxCertDelegStaking V2.StakingCredential Delegatee
+  | -- | Register and delegate staking credential to a Delegatee in one certificate. Noter that
+    -- deposit is mandatory.
+    TxCertRegDeleg V2.StakingCredential Delegatee V2.Value
+  | -- | Register a DRep with mandatory deposit value and an optional Anchor
```

Comment:
Same reason, we can't register on unregister a Ptr or a NullAddr :slightly_smiling_face: 
```suggestion
    TxCertUnRegStaking V2.Credential (Haskell.Maybe V2.Value)
  | -- | Delegate staking credential to a Delegatee
    TxCertDelegStaking V2.Credential Delegatee
  | -- | Register and delegate staking credential to a Delegatee in one certificate. Noter that
    -- deposit is mandatory.
    TxCertRegDeleg V2.Credential Delegatee V2.Value
  | -- | Register a DRep with mandatory deposit value and an optional Anchor
```

I think confusion comes from the naming. We in Ledger call this `StakeCredential`, while this thing that is called `StakingCredential` in Plutus we actually call `StakeReference`:


```haskell
data StakeReference c
  = StakeRefBase !(StakeCredential c)
  | StakeRefPtr !Ptr
  | StakeRefNull
```